client/java/src/main/java/io/openlineage/client/transports/EventTransformer.java

```diff
@@ -0,0 +1,51 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.client.transports;
+
+import io.openlineage.client.OpenLineage.DatasetEvent;
+import io.openlineage.client.OpenLineage.JobEvent;
+import io.openlineage.client.OpenLineage.RunEvent;
+import java.util.Map;
+
+/**
+ * EventTransformer is an interface that defines the transformation of an event. An implementation
+ * of this interface is required to instantiate transform interface.
+ *
+ * <p>Class implementing it needs to provide no-arg constructor
+ */
+public interface EventTransformer {
+
+  /**
+   * Initialize the transformer with properties.
+   *
+   * @param properties
+   */
+  void initialize(Map<String, String> properties);
```

Comment:
does not feel like a part of an interface to me, breaks contract behavior principle, such a method fits better to abstract class